redbot/cogs/warnings/warnings.py

```diff
@@ -342,36 +341,48 @@ def __init__(self, bot: Red):
 
     @commands.command()
     @commands.guild_only()
+    @checks.admin()
     async def warnings(
         self, ctx: commands.Context, user: Optional[Union[discord.Member, int]] = None
     ):
-        """List the warnings for the specified user.
-
-        Omit `<user>` to see your own warnings.
-
-        Note that showing warnings for users other than yourself requires
-        appropriate permissions.
-        """
+        """List the warnings for the specified user."""
         if user is None:
-            user = ctx.author
-        else:
-            if not await is_admin_or_superior(self.bot, ctx.author):
-                return await ctx.send(
-                    warning(_("You are not allowed to check warnings for other users!"))
+            await ctx.send(_("Please provide a user you want to check the warnings for."))
```

Comment:
This should either return here, or be removed entirely and remove the optional nature of the user argument (probably the latter, but either would be fine)

Reply:
Would removing the "Optional" part of the argument fix this?

Reply:
Yes, making the argument required and removing handling for `user is None` would fix this.

Reply:
Yes, making the argument required and removing handling for `user is None` would fix this.
(adding again, cause I accidentally removed the comment)